lib/Conversion/MooreToCore/MooreToCore.cpp

```diff
@@ -825,12 +825,30 @@ struct DynExtractOpConversion : public OpConversionPattern<DynExtractOp> {
                                      op->getLoc());
 
       if (isa<hw::ArrayType>(resultType)) {
+        // FIXME: for packed arrays, the part that is OOB must be 0 or X and the
+        // part overlapping with the range must match the values of the range;
+        // for unpacked arrays, if part of the accessed slice is OOB the entire
+        // result is 0 or X
         rewriter.replaceOpWithNewOp<hw::ArraySliceOp>(op, resultType,
                                                       adaptor.getInput(), idx);
         return success();
       }
 
-      rewriter.replaceOpWithNewOp<hw::ArrayGetOp>(op, adaptor.getInput(), idx);
+      Value size = rewriter.create<hw::ConstantOp>(
+          op.getLoc(), adaptor.getLowBit().getType(), arrType.getNumElements());
```

Comment:
That can cause problems if the lowBit bitwidth is too small to hold the array size.

Reply:
Nice catch. In that case we might be guaranteed to always access in bounds?

Reply:
Yes 👍 
But I need to make sure this IR is not built in that case because the pass would crash.